cli/src/pcluster/models/cluster_resources.py

```diff
@@ -172,15 +179,33 @@ class ClusterCreationFailure:
                 "Cluster has been set to PROTECTED mode due to failures detected in static node provisioning",
             ),
         ]
+        general_failure = ClusterCreationFailure("ClusterCreationFailure", "Failed to create the cluster.", "")
+        return cluster_creation_failures, general_failure
+
+    def _update_failures(self):
+        @dataclass
+        class ClusterUpdateFailure:
+            """Represent the object holding the data of ClusterCreationFailure."""
+
+            failure_code: str
+            api_failure_reason: str
+            cfn_failure_reason: str  # cloud formation matching string
+
+        cluster_update_failures = [
+            ClusterUpdateFailure("EbsMountFailure", "Failed to mount EBS volume.", "Failed to mount EBS volume"),
```

Comment:
Question: why we don't have the same kind of failures at creation time and we just have the general "ClusterCreationFailure" message?

Reply:
I separated the failures for modularity and future additions. I anticipated that we may add more update failures in the future. But I'm open to refactor and combine them!